npm/eslint-plugin-dev/lib/index.js

```diff
@@ -299,6 +299,42 @@ module.exports = {
         'mocha/no-global-tests': 'error',
         '@cypress/dev/skip-comment': 'error',
       },
+      overrides: [{
+        files: '*.tsx',
+        parser: '@typescript-eslint/parser',
+        parserOptions: {
+          ecmaVersion: 2020, // Allows for the parsing of modern ECMAScript features
+          sourceType: 'module', // Allows for the use of imports
+          ecmaFeatures: {
+            jsx: true, // Allows for the parsing of JSX
+          },
+        },
+        settings: {
+          react: {
+            version: 'detect', // Tells eslint-plugin-react to automatically detect the version of React to use
+          },
+        },
+        plugins: [
+          '@typescript-eslint',
+          'react',
+        ],
+        rules: {
+          'no-unused-vars': 'off', // avoid interface imports to be warned against
```

Comment:
In the setup guide of this rule, [@typescript-eslint/no-unused-vars](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md) it is said on the first line that we should disable the standard rule.

If we do not, ESLint sees importing an interface to create another one, as creating an unused variable, which it is not.

```tsx
import React, { useState } from 'react'

import { SpecItem } from './SpecItem'
import { OnSelectSpec } from './SpecFileItem' // this would be warned as not used even if it is actually used
import { SearchSpec } from './components/SearchSpec'
import { makeSpecHierarchy } from './make-spec-hierarchy'
import './spec-list.scss'

interface SpecsListProps {
  selectedSpecs: string[]
  specs: Cypress.Cypress['spec'][]
  onSelectSpec: OnSelectSpec 
}
```